docs/source/install.rst

```diff
@@ -0,0 +1,350 @@
+
+
+.. _prerequisites:
+
+Prerequisites
+-------------
+
+On Linux the only prerequisite needed to install Firedrake is a suitable version
+of Python (3.10 or greater). On macOS it is important that homebrew_ is installed
+and that the homebrew-installed Python is used instead of the system one.
+
+
+.. _firedrake_configure:
+
+firedrake-configure
+-------------------
+
+To simplify the installation process, Firedrake provides a utility script called
+``firedrake-configure``. This script can be downloaded by executing::
+
+  $ curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-configure
+
+
+.. _firedrake_archs:
+
+Prepared configurations
+~~~~~~~~~~~~~~~~~~~~~~~
+
+``firedrake-configure`` provides a number of different possible configurations
+(termed 'ARCH's) that specify how PETSc is configured and which external
+packages are built. The currently supported ARCHs are:
+
+* ``default``: the default installation, suitable for most users
+* ``complex``: an installation where PETSc is configured using complex numbers
+
+The different configurations can be selected by passing the flag ``--arch`` to
+``firedrake-configure``. For example::
+
+  $ python3 firedrake-configure --show-system-packages --arch complex
+
+If ``--arch`` is not specified then ``default`` is used.
+
+
+.. _install_system_dependencies:
+
+Installing system dependencies
+------------------------------
+
+If on Ubuntu or macOS, system dependencies can be installed with
+``firedrake-configure``. On Ubuntu run::
+
+  $ sudo apt install $(python3 firedrake-configure --show-system-packages)
+
+which will install the following packages:
+
+.. literalinclude:: apt_deps.txt
+   :language: text
+
+If on macOS you should instead run::
+
+  $ brew install $(python3 firedrake-configure --show-system-packages)
+
+which will install the following packages:
+
+.. literalinclude:: homebrew_deps.txt
+   :language: text
+
+If you do not have one of these systems then these dependencies will need to
+be installed manually.
+
+.. note::
+   Not all the system dependencies declared by ``firedrake-configure`` have to
+   be installed at this stage. Some (e.g. HDF5, hwloc) can also be installed
+   from source by PETSc during the ``configure`` :ref:`step<install_petsc>` by
+   passing additional flags (e.g. ``--download-hdf5``, ``--download-hwloc``).
+
+
+.. _install_petsc:
+
+Installing PETSc
+----------------
+
+For Firedrake to work as expected, it is important that a specific version of PETSc_
+is installed with a specific set of external packages. To install PETSc you need to
+do the following steps:
+
+#. Clone the PETSc repository::
+
+   $ git clone https://github.com/firedrakeproject/petsc.git
+   $ cd petsc
+
+#. Run PETSc ``configure``, passing in the flags generated by ``firedrake-configure``::
+
+   $ ./configure $(python3 ../firedrake-configure --show-petsc-configure-options)
+
+#. Compile PETSc::
+
+   $ make
```

Comment:
Need to tell the user to set PETSC_DIR and PETSC_ARCH?

Reply:
PETSc will print out a line like:
```
Configure stage complete. Now build PETSc libraries with:
   make PETSC_DIR=/path/to/petsc PETSC_ARCH=arch-firedrake-default all
```
In our installation docs could we say something along the lines of:
"PETSc will provide the required command to finish building the libraries, run this before moving on to the next step. The command will look something like ..."
and include the `configure` output above.

Reply:
will do

Reply:
done